commands/command_clone.go

```diff
@@ -19,11 +20,28 @@ var (
 	cloneFlags git.CloneFlags
 
 	cloneSkipRepoInstall bool
+
+	// cloneIsDeprecated marks whether or not the clone command is
+	// deprecated. It is false until Git v2.15.0 is released, including the
+	// 'delay' capability.
+	cloneIsDeprecated = false
 )
 
 func cloneCommand(cmd *cobra.Command, args []string) {
 	requireGitVersion()
 
+	if cloneIsDeprecated {
+		msg := []string{
```

Comment:
Should this check the git version, only printing the warning if they have 2.15.0+?

Reply:
Great idea, I added this (and an integration test for it) in: 
https://github.com/git-lfs/git-lfs/pull/2526/commits/18a3ec2041c898141b4a58ad63c6ee201179a7af